Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,96 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            bool nonExistantCheckWarning = false;
+
+            try
```

Comment:
Is there a reason why we are `try`-`catch`ing all the checks, rather than perform the check more safely?

Reply:
My reason is that as this will never fail catastrophically, which I found to be important as there would not be many other ways of constructing `PlanarSurface`s.

The other option would be to exclude all Nurbs (and Ellipses, maybe more) from the checks. which could change both in the good way of them being supported or the bad way where another curve type is added which would break everything.

Not super-sure but went with the way I knew would work.

Reply:
Personally I do not like brute forcing `try/catch` on one of the core `Create` methods..

Instead I would prefer skipping checks for `NurbsCurve` and recording a warning. Fail safe + it will enable creating nurbs-based `PlanarSurfaces`.

Reply:
What is the actual reason we are doing the try catch? 
In case the methods throw an exception, I'd resort to changing those to recording an error and returning the safest condition, e.g. `IsSelfIntersecting(NurbsCurve curve) { RecordError("NotImplemented"); return True`

Reply:
I do not know why we have so many methods with exceptions, but felt out of scope to resolve here.